### ISSUE TITLE
.astype("object") broken

### ISSUE THREAD

**Issue by jonashaag** [pandas-dev/pandas-stubs#580]
**Describe the bug**
See below.

**To Reproduce**

```py
df = pd.DataFrame({"x": [42]})
df["x"] = df["x"].astype("str")     # ok
df["x"] = df["x"].astype("object")  # fails
```

**Please complete the following information:**

Version 1.5.3.230304

**Comment by gandhis1** [comment]
I thought someone recently fixed this...maybe I'm wrong...

**Comment by Dr-Irv** [comment]
fixed in #566
